pages/text-diff.tsx

```diff
@@ -2,18 +2,99 @@ import ClearIcon from '@mui/icons-material/Clear';
 import Button from '@mui/material/Button';
+import FormControl from '@mui/material/FormControl';
+import FormHelperText from '@mui/material/FormHelperText';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
 import TextField from '@mui/material/TextField';
 import Typography from '@mui/material/Typography';
 import * as Diff from 'diff';
-import React, { useCallback, useEffect } from 'react';
+import { Change } from 'diff';
+import React, { useCallback, useEffect, useState } from 'react';
 
 import Heading from '../components/Heading';
 import Layout from '../components/Layout';
 import useLocalState from '../hooks/useLocalState';
 import useSupportsClipboardRead from '../hooks/useSupportsClipboardRead';
 
+interface DiffOptions {
+  label: string;
+  description: string;
+  value: (
+    oldString: string,
+    newString: string,
+    options?: Diff.BaseOptions | undefined,
+  ) => Change[];
+}
+
+const diffOptions: DiffOptions[] = [
+  {
+    label: 'Charaters',
+    description:
+      'Two blocks of text, comparing character by character.',
+    value: Diff.diffChars,
+  },
+  {
+    label: 'Charaters Ignore Case',
+    description:
+      'Two blocks of text, comparing character by character ignoring case.',
+    value: (oldStr: string, newStr: string) =>
+      Diff.diffChars(oldStr, newStr, { ignoreCase: true }),
+  },
+  {
+    label: 'Words',
+    description:
+      'Two blocks of text, comparing word by word, ignoring whitespace.',
+    value: Diff.diffWords,
+  },
+  {
+    label: 'Words Ignore Case',
+    description:
+      'Two blocks of text, comparing word by word, ignoring case.',
+    value: (oldStr: string, newStr: string) =>
+      Diff.diffWords(oldStr, newStr, { ignoreCase: true }),
+  },
+  {
+    label: 'Words with Space',
+    description:
+      'Two blocks of text, comparing word by word, treating whitespace as significant.',
+    value: Diff.diffWordsWithSpace,
+  },
+  {
+    label: 'Trimmed Lines',
+    description: 'Two blocks of text, comparing line by line.',
+    value: Diff.diffTrimmedLines,
+  },
+  {
+    label: 'Sentences',
+    description: 'Two blocks of text, comparing sentence by sentence.',
+    value: Diff.diffLines,
+  },
+  {
+    label: 'CSS',
+    description: 'Two blocks of text, comparing CSS tokens',
+    value: Diff.diffCss,
+  },
+  {
+    label: 'JSON',
+    description:
+      'Two JSON objects, comparing the fields defined on each. The order of fields, etc does not matter in this compparsion.',
+    value: Diff.diffJson,
+  },
+  // {
+  //   label: 'Arrays',
+  //   description:
+  //     'Two arrays, comparing each item for strict equality (===).',
+  //   value: Diff.diffArrays,
+  // },
+];
 export default function TextDiffPage() {
   const supportsClipboardRead = useSupportsClipboardRead();
+  const [selectedOptions, setSelectedOptions] = useState<
```

Comment:
Lets use `useLocalState` insetad of `useState` here to keep the option saved locally